kamon-akka-2.4.x/src/main/scala/kamon/akka/instrumentation/ActorMonitor.scala

```diff
@@ -1,22 +1,47 @@
+/*
+ * =========================================================================================
+ * Copyright © 2013-2018 the kamon project <http://kamon.io/>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ * =========================================================================================
+ */
+
+
 package akka.kamon.instrumentation
 
+import java.io.Closeable
+
 import akka.actor.{ActorCell, ActorRef, ActorSystem, Cell}
 import akka.dispatch.Envelope
 import akka.kamon.instrumentation.ActorMonitors.{TracedMonitor, TrackedActor, TrackedRoutee}
 import kamon.Kamon
 import kamon.akka.Metrics
 import kamon.akka.Metrics.{ActorGroupMetrics, ActorMetrics, RouterMetrics}
+import kamon.context.Storage.Scope
 import kamon.trace.Span
-import kamon.util.Clock
 import org.aspectj.lang.ProceedingJoinPoint
 
+final case class Traveler(envelopeContext: TimestampedContext, closeable: Closeable, timestampBeforeProcessing: Long = 0L)
```

Comment:
I'm a bit torn about this `Traveler` idea.. The biggest problem I see here is allocating the `Traveler` instances for every message we process, that seems unnecessary. Have you thought of just using mutable variables on the `ActorMonitor` implementation itself to pass these values from start to end of processing a message?

Other than that, minor nitpicking: `Traveler` doesn't really tell much about what the instance actually does. And, why pass a `Closeable` if it should simply be a `Span` instance?

Reply:
@damdev @ivantopo I think that we can avoid all the extra allocations if we use @Local annotation... The side effect is that we need rewrite some advisors in java :-(

Reply:
Yes, we was talking about that with @ivantopo i will try to follow that way